### ISSUE TITLE
fix(model): V4.1 MoE admission refuses ffn.gate.weight because the artifact's router ffn_gate_inp.weight falls through to the Llama base map

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13266]
# fix(model): V4.1 MoE admission refuses `ffn.gate.weight` because the artifact's router `ffn_gate_inp.weight` falls through to the Llama base map (streamed-dense arm)

```routing
lane: model
paths: ["internal/ggufload/deepseek41.go", "internal/model/v41_forward.go"]
class: dev
priority: P1
expected_steps: 2
```

centrality: Enabling (the native V4.1 Flash 30/5 physical witness on one Halo appliance)
p1: advanced - with the grouped `attn.wo_a` admission on trunk (`c1f811096`) the physical serve now CLEARS the layer-0 attention stage and advances into the layer-0 MoE stage, then fails closed at a NEW, distinct seam
p2: preserved - the refusal is a named `ErrV41ForwardStage` carrying the exact tensor name, not a silent mis-shape
p3: preserved - no roofline gate lowered; the vcruz Q2_K artifact `58d8ac86` stays pinned
p4: preserved - the streamed-dense bound (`FAK_STREAM_Q4K=1`) and the #13255/#13258/#13262/#13264 admissions are unchanged

## Working spine
Prompt ids -> V4.1 `forwardV41` admission -> layer-0 MoE router gate admission -> first token / new named refusal. This leaf unblocks exactly one rung: the router gate (`ffn_gate_inp`) name+shape resolution of the published artifact.

## Current state

At public trunk `c1f811096`, a physical strix3 serve of the exact staged vcruz Q2_K artifact `58d8ac86` now:

- passes the memory plan under `FAK_STREAM_Q4K=1`: `arm=streamed(resident-bound=42.768GiB) backend=vulkan staging-host-charge=103.449GiB host-budget=47.520GiB`;
- binds the gateway: `planner=inkernel engine=inkernel`;
- clears the V4.1 head stage, the per-layer norm admission (#13255), the layer-0 mHC mix stage (#13258/#13262), and the layer-0 attention output-projection admission (#13264) -- **0** `attn.wo_a` and **0** `mhc.mixes` occurrences in the log;
- enters the layer-0 MoE stage and fails closed:

```
panic: model: V4.1 forward stage moe layer=0: model: DeepSeek V4.1 forward stage is not
implemented: missing tensor model.layers.0.ffn.gate.weight
```

Call path: `internal/model/v41_forward.go:1475 prefillV41` -> `v41ForwardAdmitted` -> the `ffn.gate.weight` `v41AdmitShape` row (`v41_forward.go:685`, admit `cfg.NumExperts` x `H`).

## Why now
The grouped output projection now executes on silicon (the #13264 admission landed as `c1f811096`), so the physical serve advances into the MoE stage for the first time. The very next rung is the router gate resolution.

## The geometry discriminating question

The artifact's router gate tensor is `blk.0.ffn_gate_inp.weight` with stored dims `[5120, 384]` (H=5120, NumExperts=384; type 30). `deepseek41CanonicalSuffix` (`internal/ggufload/deepseek41.go`) has **no** `ffn_gate_inp.weight` entry, so it falls through to the shared base map, which rewrites it to the deepseek2/glm spelling `model.layers.<L>.mlp.gate.weight` (`gguf_glm_tensors.go:57`: `glmGGUFRouter = "ffn_gate_inp.weight" // mlp.gate.weight`). The native non-MLA V4.1 forward instead reads `ffn.gate.weight` (`v41_forward.go:685` admission, `:1114` read, `:1381` `matRows(wGate, xn, cfg.NumExperts, H)`).

So the loader name-resolution gap is the same class as #13255: the artifact carries the tensor, but under a name the native forward does not read. The forward's admitted shape is `[NumExperts, H]`; the artifact stores `[5120, 384]` = `[H, NumExperts]`, so the admission/read likely also needs the transposed-storage form (as the shared-experts sibling `ffn_gate_shexp.weight` does: mapped with an explicit "admit [I, H]" comment while the artifact stores `[5120, 2304]` = `[H, I]`).

Gather first: dump the artifact's `ffn_gate_inp.weight` / `exp_probs_b.bias` dims on layer 0 and every block, map them onto `v41Route` (`v41_forward.go:1382`) and the reference `inference/model.py` DeepseekV4 router before changing the loader map. Do NOT guess a transpose that silently mis-shapes the router logits.

## Core through-line
Prompt ids -> V4.1 `forwardV41` admission -> layer-0 MoE router gate -> first token / new named refusal.

## Gold-plating boundary
- Any change to the routing math (`v41Route`), expert grouping, or the MoE arithmetic itself.
- Lowering any roofline gate, reshaping the mHC/CED/attention geometry, or touching the cache lanes.
- Speculative caching, multi-node replication, or premature optimization before the single-node execution receipt.

## Done condition

Definition of done:
- [ ] A regression test pins the `ffn.gate.weight` router-gate seam red on the parent commit and green after the fix.
- [ ] `go test ./internal/model/ -count=1` and `go test ./internal/ggufload/ -count=1` pass.
- [ ] `go run ./cmd/fak-boundary check --all` exits 0 with 0 violations.
- [ ] The physical strix3 serve at the post-fix trunk binary advances past the layer-0 MoE router gate (or reaches a new named refusal), captured in a durable receipt.

## Work estimate
Estimate: 2 points (expected_steps: 2).

## Parent context
#12640 (native DeepSeek V4.1 Flash execution and physical qualification). Sibling seam issues: #13255 (per-layer norm, CLEARED on silicon), #13258 (mHC mix admission geometry, CLEARED), #13262 (mHC mix residency-complete read, CLEARED on silicon at `1e97b2ef1`), #13245 (MLA attn_kv width, CLEARED), #13264 (grouped `attn.wo_a` admission, CLEARED on silicon at `c1f811096`).

## Acceptance gate
`go test ./internal/model/ -run 'V41|MoE|Gate|Route' -count=1` and `go test ./internal/ggufload/ -count=1` must pass, `go vet`/`gofmt` clean, and `go run ./cmd/fak-boundary check --all` exits 0. The regression test must be RED on the parent commit and GREEN after the fix.

## Closure binding
This issue closes only when the physical strix3 serve at the post-fix trunk binary advances past the layer-0 MoE router gate (or reaches a new named refusal) AND a `[HW-WITNESSED]` receipt is posted in the closing comment. A green software test alone does NOT close it.

## Witness
[HW-WITNESSED] on physical strix3 at trunk `c1f811096`, exclusive lease `lease-strix3-abe43ed2fcb2` (T2 30m, holder `mission-native-13264-hw`), resident `fak.service` stopped (56 GiB free) and restored `active` after. Binary `fak-vulkan-c1f811096-clean` sha256 `89a6f0910b6cff278472e4c524588caac1d9340f00147578233b72ecdc8e081f` (`vcs.revision=c1f8110968d05ce1107643047bc8515fd0d67a5a`, `vcs.modified=false`, `-tags vulkan`). Log `/home/fak/v41-witness/serve-c1f811096.log` sha256 `198b75772a305c673ee20c7a6859da29e81872fdd4516c0b752f12b1d52750b5`; 1 `ffn.gate` occurrence, 0 `attn.wo_a`, 0 `mhc.mixes`, and 0 `Out of memory: Killed process` from this run (host dmesg OOM lines timestamp 05:06-07:07, this run 16:16 local; run exited status=2/INVALIDARGUMENT, not SIGKILL). No token generated; the 30/5 first-witness bar remains unchecked.

## Overall completion contribution
Contribution: 2/2 points.

## Completion standard
production

## Target operating envelope
- host_available_gib: >= 50 GiB

## Witnessed operating envelope
- host_available_gib: >= 56 GiB

**Comment by anthony-chaudhary** [comment]
Landed on public trunk: **fe6b49f8e** ((fak model)), rebased on 8d10967e7.

## What changed
The V4.1 router gate resolved through the SHARED deepseek2/GLM batched-MoE branch of CanonicalTensorNameArch (rchUsesGGUFBatchedMoEExperts includes deepseek41), which rewrites lk.<L>.ffn_gate_inp.weight to model.layers.<L>.mlp.gate.weight. The native non-MLA V4.1 forward reads model.layers.<L>.ffn.gate.weight (41_forward.go:685 admission, :1114 read, :1381 matRows(wGate, xn, cfg.NumExperts, H)), so 41AdmitShape refused a tensor the artifact DOES carry.

Fix (2 files, name resolution only):
- internal/ggufload/gguf_tensor_canonical.go: exclude deepseek41 from the shared batched-MoE router branch.
- internal/ggufload/deepseek41.go: map fn_gate_inp.weight -> ffn.gate.weight.

41Route arithmetic, the admitted [E, H] shape, and every other arch (deepseek2, glm_moe_dsa, llama, qwen2) are unchanged.

## Software witness (SW-VERIFIED)
- New regression deepseek41_router_gate_seam_test.go: **RED on parent** 5fa541a4 (returned model.layers.<L>.mlp.gate.weight), **GREEN after** the fix.
- go test ./internal/ggufload/ -count=1 => ok (17.1s)
- go test ./internal/model/... -count=1 => ok (internal/model 178s, internal/model/v41 ok)
- go vet ./internal/ggufload/ ./internal/model/ => clean; gofmt -l => clean
- go run ./cmd/fak-boundary check --staged => PASS, 0 violations across all 5 gates

## Remaining rung ([HW-WITNESSED], still OPEN)
Per the closure binding, this issue stays open until a physical strix3 serve at a post-fix trunk binary advances past the layer-0 MoE router gate (or reaches a new named refusal) with a [HW-WITNESSED] receipt. That hardware rung was not available in this bounded lane.

Ref #12640
